services/user-svc/internal/organization/organization.go

```diff
@@ -341,6 +342,77 @@ func (s ServiceServer) AcceptInvitation(ctx context.Context, req *pb.AcceptInvit
 	return &pb.AcceptInvitationResponse{}, nil
 }
 
+func (s ServiceServer) GetInvitationsByUser(ctx context.Context, req *pb.GetInvitationsByUserRequest) (*pb.GetInvitationsByUserResponse, error) {
+
+	db := hwgorm.GetDB(ctx)
+	log := zlog.Ctx(ctx)
+
+	claims, err := common.GetAuthClaims(ctx)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	var invitations []Invitation
+	var invitationsResponse []*pb.GetInvitationsByUserResponse_Invitation
+
+	filter := db.Where("email = ?", claims.Email)
+	if req.State != nil {
+		filter = filter.Where("state = ?", req.State)
+	}
+
+	if err := filter.Find(&invitations).Error; err != nil {
+		if hwgorm.IsOurFault(err) {
+			log.Warn().Err(err).Msg("database error")
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+	}
+
+	invitationsResponse = hwutil.Map(invitations, func(invitation Invitation) *pb.GetInvitationsByUserResponse_Invitation {
+		return &pb.GetInvitationsByUserResponse_Invitation{
+			Id:             invitation.ID.String(),
+			Email:          invitation.Email,
+			OrganizationId: invitation.OrganizationID.String(),
+			State:          invitation.State,
+		}
+	})
+
+	return &pb.GetInvitationsByUserResponse{
+		Invitations: invitationsResponse,
+	}, nil
+}
+
+func (s ServiceServer) DeclineInvitation(ctx context.Context, req *pb.DeclineInvitationRequest) (*pb.DeclineInvitationResponse, error) {
```

Comment:
Check of mail and state transition as in `AcceptInvitation()` is missing. We should wrap the query of an invitation with a function called `GetInvitationByIdAndEmail()`, the state check should be a part of the controller due to the error handling. After that, we could reuse this function for `AcceptInvitation()` as well.